modules/distr-tracing-tempo-config-query-frontend.adoc

```diff
@@ -6,62 +6,125 @@
-
-|spec:
-  query:
-    replicas:
-|Specifies the number of Query replicas to create.
-|Positive integer
-|
+|Parameter |Description |Values
+
+|`nodeSelector`
+|The simple form of the node-selection constraint.
+|type: object
+
+|`replicas`
+|The number of replicas to be created for the component.
+|type: integer; format: int32
+
+|`tolerations`
+|Component-specific pod tolerations.
+|type: array
 |===
 
+The query frontend component is responsible for sharding the search space for an incoming query. Traces are exposed via a simple HTTP endpoint: `GET /api/traces/<traceID>`. Internally, the query frontend component splits the `blockID` space into a configurable number of shards and then queues these requests. The querier component connects to the query frontend component via a streaming gRPC connection to process these sharded queries.
 
-.Configuration parameters passed to Query
+.Configuration parameters for the query frontend component
 [options="header"]
-[cols="l, a, a, a"]
 |===
-|Parameter |Description |Values |Default value
-
-|spec:
-  query:
-    options: {}
-|Configuration options that define the Query service.
-|
-|
-
-|options:
-  log-level:
-|Logging level for Query.
-|`debug`, `info`, `warn`, `error`, `fatal`, `panic`
-|
-
-|options:
-  query:
-    base-path:
-|You can set the base path for all tempo-query HTTP routes to a non-root value: for example, `/tempo` will cause all UI URLs to start with `/tempo`. This can be useful when running `tempo-query` behind a reverse proxy.
-|`/<path>`
-|
+|Parameter |Description |Values
+
+|`component`
+|Configuration of the query frontend component.
+|type: object
+
+|`component.nodeSelector`
+|The simple form of the node selection constraint.
+|type: object
+
+|`component.replicas`
+|The number of replicas to be created for the query frontend component.
+|type: integer; format: int32
+
+|`component.tolerations`
+|Pod tolerations specific to the query frontend component.
+|type: array
+
+|`jaegerQuery`
+|The options specific to the Jaeger Query component.
+|type: object
+
+|`jaegerQuery.enabled`
+|When `enabled`, creates the Jaeger Query component,`jaegerQuery`.
+|type: boolean
+
+|`jaegerQuery.ingress`
+|The options for the Jaeger Query ingress.
+|type: object
+
+|`jaegerQuery.ingress.annotations`
+|The annotations of the ingress object.
+|type: object
+
+|`jaegerQuery.ingress.host`
+|The hostname of the ingress object.
+|type: string
+
+|`jaegerQuery.ingress.ingressClassName`
+|The name of an IngressClass cluster resource. Defines which ingress controller serves this ingress resource.
+|type: string
+
+|`jaegerQuery.ingress.route`
+|The options for the OpenShift route.
+|type: object
+
+|`jaegerQuery.ingress.route.termination`
+|The termination type. The default is `edge`.
+|type: string (enum: insecure, edge, passthrough, reencrypt)
+
+|`jaegerQuery.ingress.type`
+|The supported type of ingress for the Jaeger Query UI: `ingress`, `route`, and `none`.
```

Comment:
I'd expands this a bit to make sure it's clear:

The type of ingress for the Jaeger Query UI. The supported types are `ingress`, `route`, and `none`.

Reply:
This /lgtm as is